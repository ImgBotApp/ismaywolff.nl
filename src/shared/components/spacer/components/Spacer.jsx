import styled from 'styled-components'

const setMargin = direction => props => (props[direction] ? props[direction] : 0)

const Spacer = styled.div`
  /* stylelint-disable */
  margin-top: ${setMargin('top')};
  margin-bottom: ${setMargin('bottom')};
  margin-left: ${setMargin('left')};
  margin-right: ${setMargin('right')};
  /* stylelint-enable */
`

export default Spacer
