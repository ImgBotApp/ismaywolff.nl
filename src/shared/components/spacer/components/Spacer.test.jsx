/**
 * @jest-environment node
 */

import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Spacer from './Spacer'

describe('<Spacer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Spacer />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('renders top correctly', () => {
    const wrapper = shallow(<Spacer top="1rem" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('renders left correctly', () => {
    const wrapper = shallow(<Spacer left="1rem" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('renders right correctly', () => {
    const wrapper = shallow(<Spacer right="1rem" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('renders bottom correctly', () => {
    const wrapper = shallow(<Spacer bottom="1rem" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })
})
