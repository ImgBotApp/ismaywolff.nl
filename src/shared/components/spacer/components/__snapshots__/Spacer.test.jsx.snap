// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> renders bottom correctly 1`] = `
.cfrKdS {
  margin-top: 0;
  margin-bottom: 1rem;
  margin-left: 0;
  margin-right: 0;
}

<div
  className="sc-bdVaJa cfrKdS"
/>
`;

exports[`<Spacer /> renders correctly 1`] = `
.dJmyIz {
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
}

<div
  className="sc-bdVaJa dJmyIz"
/>
`;

exports[`<Spacer /> renders left correctly 1`] = `
.kGSnBO {
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 1rem;
  margin-right: 0;
}

<div
  className="sc-bdVaJa kGSnBO"
/>
`;

exports[`<Spacer /> renders right correctly 1`] = `
.cFxmSh {
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 1rem;
}

<div
  className="sc-bdVaJa cFxmSh"
/>
`;

exports[`<Spacer /> renders top correctly 1`] = `
.hXpbAn {
  margin-top: 1rem;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
}

<div
  className="sc-bdVaJa hXpbAn"
/>
`;
